EXAMPLES.md

```diff
@@ -197,3 +197,121 @@ class AcsRespondController
+ 
+### Code example 1: setting up security.yaml
+
+```yaml
+security:
+  # Other security entries are not shown in this code example. Only the required entries for getting SAML 
+  # authentications with the stepup-saml-bundle are stated below
+
+  access_control:
+    - { path: ^/saml, roles: IS_AUTHENTICATED_ANONYMOUSLY, requires_channel: https }
+    - { path: ^/, roles: IS_AUTHENTICATED_FULLY, requires_channel: https }
+  
+  providers:
+    saml-provider:
+      id: YourApp\Security\Authentication\Provider\SamlProvider
+      
+  firewalls:
+    login_firewall:
+      pattern:    ^/saml/metadata
+    
+    saml_based:
+      custom_authenticators:
+        - Surfnet\SamlBundle\Security\Authentication\SamlAuthenticator
+```
+
+### Code example 2: services.yaml
+```yaml
+
+    # This is just an example of the dependencies a SamlProvider could have, taylor this provider to your own needs
+    surfnet_saml.saml_provider:
+        class: YourApp\Security\Authentication\Provider\SamlProvider
+        arguments:
+            - '@YourApp\Repository\UserRepository'
+            - '@surfnet_saml.saml.attribute_dictionary'
+            - '@logger'
+
+    # Be sure to create an alias to the `surfnet_saml.saml_provider`, otherwise Symfony will not be able to autowire the
+    # `security.listener.user_provider` service.
+    Surfnet\ServiceProviderDashboard\Infrastructure\DashboardSamlBundle\Security\Authentication\Provider\SamlProvider:
+        alias: surfnet_saml.saml_provider
+
+```
+
+### Code example 3: example failure handler
+```php
+<?php
+
+/**
+ * Copyright 2023 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace YourApp\Security\Authentication\Handler;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Http\Authentication\DefaultAuthenticationFailureHandler;
+use Symfony\Component\Security\Http\HttpUtils;
+use Twig\Environment;
+
+class FailureHandler extends DefaultAuthenticationFailureHandler
+{
+    public function __construct(
+        HttpKernelInterface $httpKernel,
+        HttpUtils $httpUtils,
+        array $options = [],
+        LoggerInterface $logger = null,
+        private readonly Environment $templating,
```

Comment:
Readonly properties were introduced in PHP 8.1, but this library still supports 8.0.. I think the example should be 8.0 compatible

Reply:
Good point, will remove the `readonly` keywords from the rest of the project for now.